tokio/src/sync/mutex.rs

```diff
@@ -145,12 +165,20 @@ fn bounds() {
     // This has to take a value, since the async fn's return type is unnameable.
     fn check_send_sync_val<T: Send + Sync>(_t: T) {}
     fn check_send_sync<T: Send + Sync>() {}
+    fn check_static<T: 'static>() {}
+    fn check_static_val<T: 'static>(_t: T) {}
+
     check_send::<MutexGuard<'_, u32>>();
+    check_send::<OwnedMutexGuard<u32>>();
```

Comment:
We should consider moving this to `tests/async_send_sync.rs`, which is not only able to detect missing `Send` impls, but also able to detect if something is `Send` when it shouldn't be. The file currently has mostly async functions, but [it can handle types too][1].

[1]: https://github.com/tokio-rs/tokio/blob/a26d3aec961fbb748217f853ba33af7007736b8b/tokio/tests/async_send_sync.rs#L247